templates/cisco_ios_show_tacacs.template

```diff
@@ -0,0 +1,4 @@
+Value TACACS (\d+\.\d+\.\d+\.\d+)
+
+Start
+  ^\s+Server address:\s+${TACACS} -> Record
```

Comment:
The last line in the captures should be `^.*` -> Error`, so that we can ensure we know the data. I know that most of them don't have this, but it is the standard for all new templates. This also means that all of the lines that are not capturing data still need to be matched. They don't have to be elaborate, but something like `^\s+Socket` would match several lines and prevent from hitting the `Error` line.

Reply:
Not quite sure to understand this. If I add the line as you said, the script iterate over just the first tacacs output and is not catching the port

```Value TACACS (\d+\.\d+\.\d+\.\d+)
Value PORT (\d+)

Start
  ^\s+Server address:\s+${TACACS}
  ^\s+Server port:\s+${PORT}^.* -> Error```

TASK [debug] ******************************************************************************************************************************************************************************************************
ok: [IE2BBRTR02] => {
    "show_tacacs": [
        {
            "PORT": "", 
            "TACACS": "10.40.9.140"
        }
    ]
}

```

Reply:
Look at this one: https://github.com/jmcgill298/ntc-templates/blob/9c6df74b6a2f30c3427aa6f43aa55ba6e57f45b8/templates/hp_procurve_show_vlans.template

The last line should literally be `^.* -> Error`
```
  ^\s+Server address:\s+${TACACS}
  ^\s+Server port:\s+${PORT} -> Record
  ^$match3 #this needs to be updated to a regex line, and you will probably need multiple of these
  ^\s*$$
  ^.* -> Error
```

Reply:
what  '^$match3' should match?

Reply:
you need to match every line in the raw data, or else it will hit the `Error` and raise an exception. If you just add ^.* -> Error to your current template and run it, you will see what I mean.

A line like `^\s+Socket` would account for four of the lines in the raw sample that you are currently not matching; that leaves 3 additional lines that you would need to match to avoid hitting the Error